include/core/CStatistics.h

```diff
@@ -138,8 +151,40 @@ class CORE_EXPORT CStatistics : private CNonCopyable {
     //! Collection of statistics
     TStatArray m_Stats;
 
+    //! cache of stats for use in persistence
+    TStatsCacheUPtr m_Cache;
```

Comment:
Couldn't this just be `std::unique_ptr<TStatsArray>`? I'm not sure CStatsCache is adding anything.